include/core/utility.h

```diff
@@ -1,29 +1,30 @@
 // This file is subject to the terms and conditions defined in 'LICENSE' in the source code package
 
-#ifndef JACTORIO_CORE_UTILITY_H
-#define JACTORIO_CORE_UTILITY_H
+#ifndef JACTORIO_INCLUDE_CORE_UTILITY_H
+#define JACTORIO_INCLUDE_CORE_UTILITY_H
 #pragma once
 
-#include <algorithm>
 #include <string>
 
+#include "jactorio.h"
+
 namespace jactorio::core
 {
     ///
-    /// \brief Converts provided string to lower case
-    inline std::string StrToLower(const std::string& str) {
-        std::string temp = str;
-        std::transform(temp.begin(), temp.end(), temp.begin(), tolower);
-        return temp;
-    }
+    /// Converts provided string to lower case
+    J_NODISCARD std::string StrToLower(std::string str);
+
+    ///
+    /// Converts provided string to lower case
+    J_NODISCARD std::string StrToUpper(std::string str);
+
+    ///
+    /// Remove leading and trailing whitespace ' ' and '\0'
+    J_NODISCARD std::string StrTrimWhitespace(std::string str);
```

Comment:
Make such that it takes characters to be recognized as whitespace as a parameter